Backend/routes/auth.js

```diff
@@ -45,7 +48,7 @@ router.post("/register", async (req, res) => {
       return res.status(500).json({ message: "Internal Server Error" });
```

Comment:
Try not to return `500` Error from backend at any point, its not a good practice.
It can be (or something more relevant):
```suggestion
      return res.status(400).json({ message: "Bad request." });
```